### DISCUSSION TITLE
Using `one` to serialize an active record serialized field

### DISCUSSION THREAD

**Discussion by bernardeli** [okuramasafumi/alba#203] [Q&A] [answered]
Hello there,

I chose to use Alba after evaluating many other options. So first, I want to say a big thank you.

I'm currently developing a new API for a resource that contains many ActiveRecord serialized types, such as a [money](https://github.com/RubyMoney/money-rails) object.

Money creates at least two fields: `price_currency` and `price_cents`.

Let's suppose a Product object that has that fields. I got the following working but feels odd to use `one` when that is not a relationship.

```ruby
class Product < AR::Base
  monetize :price_cents
end

class MoneyResource
  include Alba::Resource

  attribute :cents do |money|
    money.cents
  end

  attribute :currency do |money|
    money.currency_as_string
  end

  attribute :formatted do |money|
    money.format
  end
end

class ProductResouce
  include Alba::Resource

  attributes :id, :name 
 
  one :price, resource: MoneyResource
end
```

I'm aiming to produce an output like the below:

```ruby
{
  id: 1,
  name: "Foo",
  price: {
    cents: 1000,
    currency: "USD",
    formatted: "$10,00",
  }
}
```

The next step is, let's say that, for historical reasons, the attribute from `Product` is not `price` but it's `suggested_price`, but through the API, I want to expose it as `price`. How can I use `one`/`many` for relationships but change the relationship name?

Again, thank you. I appreciate it if you can shed some light on this. 😊

**Comment by bernardeli** [comment]
Thinking more about that, what would you think of using the following:

```ruby
class ProductResouce
  attribute :price do |product|
    MoneyResource.new(product.suggested_price).to_h
  end
end
```

**Comment by okuramasafumi** [comment, marked as answer]
Hi!

> I got the following working but feels odd to use one when that is not a relationship.

It can be read as "Product has one price." Personally it's acceptable, but if you want to avoid that then you could use the code like you posted later.

> How can I use one/many for relationships but change the relationship name?

You can add `key` option to `one` method to change its key. See https://rubydoc.info/gems/alba/Alba/Resource/ClassMethods#association-instance_method
(Sorry that it's not in README, I'll add the mention to it)

**Reply by bernardeli** [reply]
Ohhh perfect! I did a deep dive in the code base but missed the `key` attribute. That'll help me.

Thanks for chiming in. I appreciate the input you gave me.
